cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/connector/resource/AzureDatabaseResourceService.java

```diff
@@ -148,9 +155,16 @@ public List<CloudResourceStatus> terminateDatabaseServer(AuthenticatedContext ac
             List<CloudResource> resources, boolean force, PersistenceNotifier persistenceNotifier) {
         CloudContext cloudContext = ac.getCloudContext();
         AzureClient client = ac.getParameter(AzureClient.class);
-
+        String keyVaultUrl = null;
+        String keyVaultResourceGroupName = null;
+        Map<String, Object> dbStackParams = Optional.ofNullable(stack.getDatabaseServer())
+                .map(DatabaseServer::getParameters).orElse(null);
+        if (dbStackParams != null && dbStackParams.containsKey("keyVaultUrl")) {
+            keyVaultUrl = (String) dbStackParams.getOrDefault("keyVaultUrl", null);
+            keyVaultResourceGroupName = (String) dbStackParams.getOrDefault("keyVaultResourceGroupName", null);
```

Comment:
Sanity check: Are these stack parameters already populated in the redbeams side when invoking DB server termination?

Reply:
Updated the code in `DBStackToDatabaseStackConverter.java` to handle this.